File: server/src/interfaces/author.ts
import { ObjectId } from 'mongodb';

export interface Author {
    _id: ObjectId
    name: string;
    sanitizedName: string;
    aliases: Array<string>;
    bio?: string;

    /**
     * Array of ISBNs of books written by this author. Reference to the books collection.
     */
    books: Array<string>;
}

File: java-server/src/main/java/com/mongodb/devrel/library/domain/model/Author.java
package com.mongodb.devrel.library.domain.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.util.List;

@Document(collection = "authors")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Author {

    @MongoId(FieldType.OBJECT_ID)
    @JsonProperty("_id")
    private String id;

    private String name;

    private String sanitizedName;

    private String bio;

    private List<Book> books;

    private List<String> aliases;
}
